File: lib/shape.rb
class Shape
    attr_reader :name
  
    def initialize(name)
      @name = name
    end
  end
  
  class Rectangle < Shape
    attr_reader :width, :height
  
    def initialize(width, height)
      super("Rectangle")
      @width = width
      @height = height
    end
  
    def area
      width * height
    end
  end
  
#rectangle = Rectangle.new(5, 10)
#area = rectangle.area
#uts "Area of rectangle: #{area}"

File: lib/shape.py
class Shape:
  def __init__(self, name):
    self.name = name
class Rectangle(Shape):
  def __init__(self, width, height):
    super().__init__("Rectangle")
    self.width = width
    self.height = height

  def area(self):
    return self.width * self.height

# Usage
rectangle = Rectangle(5, 10)
area = rectangle.area()
print(f"Area of rectangle: {area}")